### ISSUE TITLE
Bug: Maps  "subtype" is set to null

### ISSUE THREAD

**Issue by mwallschlaeger** [GeoNodeUserGroup-DE/contrib_datapackage#15]
to separate between type of maps (tabular-collection, map) it would be good to reuse the subtype element from dataset objects. Currently the \`subtype` is set to `null`

**Comment by ridoo** [comment]
The current approach is, to dynamically handle map routes based on their content (i.e. tabular layers only). If you want to store maps with a subtype we would have to update it when adding/removing layers.

**Comment by ridoo** [comment]
To support this, I tested some things out (on the frontend and on the backend). I came up with a solution where the backend sorts out if all map layers are of `subtype="tabular"`. Actually, it is the Serializer which already provides all dataset information from the database, so no update on the client is needed.

However, removing all spatial layers from an existing map will not trigger a redirect to `/tabular-collection` at the moment.
